### PR TITLE
[cv_bridge] Add colormap option for cvtColorForDisplay

### PR DESCRIPTION
**Pull request by wkentaro**

### CODE REVIEW

**Comment by wkentaro** [comment]
The test depends on image_view...

```
/home/travis/catkin_ws/src/image_pipeline/image_view/src/nodelets/image_nodelet.cpp: In member function ‘void image_view::ImageNodelet::imageCb(const ImageConstPtr&)’:

/home/travis/catkin_ws/src/image_pipeline/image_view/src/nodelets/image_nodelet.cpp:165:87: error: could not convert ‘do_dynamic_scaling’ from ‘bool’ to ‘cv_bridge::cvtColorForDisplay_opts’

     last_image_ = cvtColorForDisplay(cv_bridge::toCvShare(msg), "", do_dynamic_scaling)->image;

                                                                                       ^

make[2]: *** [CMakeFiles/image_view.dir/src/nodelets/image_nodelet.cpp.o] Error 1

make[1]: *** [CMakeFiles/image_view.dir/all] Error 2
```

**Comment by wkentaro** [comment]
rebased.

File: cv_bridge/include/cv_bridge/cv_bridge.h [hunk 1]
```diff
@@ -247,6 +247,15 @@ CvImageConstPtr toCvShare(const sensor_msgs::Image& source,
 CvImagePtr cvtColor(const CvImageConstPtr& source,
                     const std::string& encoding);
 
+struct cvtColorForDisplay_opts {
```
Comment (vrabaud) [inline cv_bridge/include/cv_bridge/cv_bridge.h]: cvtColorForDisplayOptions instead to keep the style coherent.

File: cv_bridge/src/cv_bridge.cpp [hunk 2]
```diff
@@ -532,12 +533,11 @@ CvImageConstPtr cvtColorForDisplay(const CvImageConstPtr& source,
       // Let's decide upon an output format
       if (enc::numChannels(source->encoding) == 1)
       {
-        if (source->encoding == enc::TYPE_32SC1)
+        if ((source->encoding == enc::TYPE_32SC1) ||
```
Comment (vrabaud) [inline cv_bridge/src/cv_bridge.cpp:536]: wait, so we always convert to BGR now ? No more grayscale ?

Comment (wkentaro) [inline cv_bridge/src/cv_bridge.cpp:536]: When colormap==-1 (this is default for compatibility with previous behavior), the encoding was force be changed to grayscale (https://github.com/ros-perception/vision_opencv/pull/130/files#diff-fb2cbb0564c5ce322e4a56960733ecf4R621). 
(the encoding was ignored), so maybe this does not change the behavior so much.
But if you are concerning about the ambiguousness of encoding, I can change it. (but I'm not sure what encoding is correct when encoding was not specified as argument for cvtColorForDisplay...)

File: cv_bridge/include/cv_bridge/cv_bridge.h (hunk 1 above)
Comment (vrabaud) [inline cv_bridge/include/cv_bridge/cv_bridge.h]: sorry, it should be CvtColorForDisplayOptions

**Comment by vrabaud** [comment]
Please fix CvtColorForDisplayOptions, squash your commits and that will be good (Travis got fixed).

**Comment by wkentaro** [comment]
Changed struct name and squashed the commits.

**Comment by vrabaud** [comment]
and Travis passes. Thx !

**Comment by ayamiya** [comment]
Changing API affect some people, for example
```
process[image_view-3]: started with pid [14449]
terminate called after throwing an instance of 'std::runtime_error'
  what():  couldn't set format
/opt/ros/indigo/lib/image_view/image_view: symbol lookup error: /opt/ros/indigo/lib/image_view/image_view: undefined symbol: _ZN9cv_bridge18cvtColorForDisplayERKN5boost10shared_ptrIKNS_7CvImageEEERKSsNS_25CvtColorForDisplayOptionsE
[image_view-3] process has died [pid 14449, exit code 127, cmd /opt/ros/indigo/lib/image_view/image_view image:=image_rect_color __name:=image_view __log:=/home/mech-user/.ros/log/e51387ac-b79c-11e6-b1b8-e8b1fce891c1/image_view-3.log].
log file: /home/mech-user/.ros/log/e51387ac-b79c-11e6-b1b8-e8b1fce891c1/image_view-3*.log
```

need to run
```
sudo apt-get update
sudo apt-get install ros-indigo-image-view ros-indigo-cv-bridge
```

This breaks example program of all student  at @k-okada 's class room

**Comment by wkentaro** [comment]
@ayamiya What do you mean?

**Comment by k-okada** [comment]
changing api breaks image-view, thy know that `apt-get update && apt-get dist-upgrade` will fix this, but many people worried to run that command because someone reported that this command troubles on his computer. So question is wy why this happened? `apt-get install ros-indigo-image-view` do not update cv-bridge if that is already installed? what happens if we write `<build_depend version_gte="1.12.0">cv_bridge</build_depend>` ?? etc...

**Comment by wkentaro** [comment]
I think the cause is not the change of api, but the write of dependency in image_view.
Currently image_view uses the latest cv_bridge with `cv_bridge::CvtColorForDisplayOptions`
https://github.com/ros-perception/image_pipeline/blob/indigo/image_view/src/nodes/image_view.cpp#L68

So adding `<run_depend version_gte="1.11.13">cv_bridge</run_depend>` to image_view should be the solution.
So an issue should be open in [ros-perception/image_pipeline](https://github.com/ros-perception/image_pipeline/issues).

BTW, you can run `apt-get update && aptitude safe-upgrade` if you worry about the upgrade of the kernel.

**Comment by wkentaro** [comment]
https://github.com/ros-perception/image_pipeline/pull/239 should solve the problem.

**Comment by k-okada** [comment]
version_gte will not change deb dependency, at least for now...
https://github.com/start-jsk/rtmros_common/blob/master/hrpsys_ros_bridge/package.xml#L38
```
$ aptitude show ros-indigo-hrpsys-ros-bridge
Package: ros-indigo-hrpsys-ros-bridge    
State: installed
Automatically installed: yes
Version: 1.3.2-0trusty-20161018-042753-0700
Priority: extra
Section: misc
Maintainer: Kei Okada <k-okada@jsk.t.u-tokyo.ac.jp>
Architecture: amd64
Uncompressed Size: 30.7 M
Depends: libboost-system1.54.0, libboost-thread1.54.0, libc6 (>= 2.14), libgcc1
         (>= 1:4.1.1), libomniorb4-1 (>= 4.1.6), libomnithread3c2 (>= 4.0.6),
         libstdc++6 (>= 4.6), ros-indigo-openhrp3, ros-indigo-openrtm-aist,
         ipython, python-psutil, ros-indigo-actionlib,
         ros-indigo-camera-info-manager, ros-indigo-collada-urdf,
         ros-indigo-control-msgs, ros-indigo-diagnostic-aggregator,
         ros-indigo-diagnostic-msgs, ros-indigo-dynamic-reconfigure,
         ros-indigo-geometry-msgs, ros-indigo-hrpsys, ros-indigo-hrpsys-tools,
         ros-indigo-image-transport, ros-indigo-nav-msgs,
         ros-indigo-pr2-controllers-msgs, ros-indigo-pr2-msgs,
         ros-indigo-robot-pose-ekf, ros-indigo-robot-state-publisher,
         ros-indigo-roscpp, ros-indigo-rosnode, ros-indigo-rostest,
         ros-indigo-rqt-gui, ros-indigo-rqt-gui-py,
         ros-indigo-rqt-robot-dashboard, ros-indigo-rqt-robot-monitor,
         ros-indigo-rtmbuild, ros-indigo-sensor-msgs, ros-indigo-tf,
         ros-indigo-visualization-msgs
Conflicts: ros-indigo-hrpsys-ros-bridge
Description: hrpsys_ros_bridge package provides basic functionalities to bind hrpsys, a OpenRTM-based controller, and ROS.
 By using this package, you can write your ROS packages that communicate with
 your non-ROS robots that run on hrpsys. For communicating with the robots that
 run on OpenRTM without hrpsys, you can use openrtm_ros_bridge package.
Homepage: http://wiki.ros.org/hrpsys_ros_bridge
```

**Comment by wkentaro** [comment]
But it should be `run_depend` instead of `build_depend`, right?

**Comment by k-okada** [comment]
yes, you're right.  run_depends for 'Depends' and build_depends for 'Build Depends'

**Pull request merged**